.github/workflows/develop.yml

```diff
@@ -0,0 +1,128 @@
+name: develop
+
+on:
+  pull_request:
+  push:
+
+jobs:
+
+  linux-x86:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Dep
+        run: |
+          rustup component add clippy
+          rustup component add rustfmt
+      # https://github.com/EmbarkStudios/cargo-deny-action
+      - uses: EmbarkStudios/cargo-deny-action@v1
+        with:
+          arguments: --all-features
+          command: check advisories licenses sources bans
+      - name: Run ci-generated
+        run: make ci-generated
+      - name: Run ci
+        run: make ci
+      - name: Run ci-asm
+        run: make ci-asm
+      - name: Run ci-asm-chaos
+        run: make ci-asm-chaos
+      - name: Run test suite
+        run: |
+          sudo apt install -y git \
+                              build-essential \
+                              autoconf \
+                              automake \
+                              autotools-dev \
+                              libmpc-dev \
+                              libmpfr-dev \
+                              libgmp-dev \
+                              gawk \
+                              libtool \
+                              patchutils \
+                              libexpat-dev \
+                              zlib1g-dev
+          git clone https://github.com/nervosnetwork/ckb-vm-test-suite
+          ln -snf .. ckb-vm-test-suite/ckb-vm
+          docker run --rm -v `pwd`:/code nervos/ckb-riscv-gnu-toolchain:bionic-20210804 cp -r /riscv /code/riscv
+          cd ckb-vm-test-suite
+          git checkout 86480364649c9cb6ac01674fe51156e7cf50a31a
+          git submodule update --init --recursive
+          RISCV=`pwd`/../riscv ./test.sh
+      - name: Run spawn tests in release
+        run: cargo test test_spawn --release --features=asm -- --nocapture
+
+  linux-arm:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Dep
+        run: |
+          sudo apt update -y
+          sudo apt install -y build-essential \
+                              autoconf \
+                              automake \
+                              autotools-dev \
+                              libmpc-dev \
+                              libmpfr-dev \
+                              libgmp-dev \
+                              gawk \
+                              libtool \
+                              patchutils \
+                              libexpat-dev \
+                              zlib1g-dev
+          rustup target add aarch64-unknown-linux-gnu
+      - name: Pre build
```

Comment:
I was slightly confused on the organization here:

* `Pre build` task is only used for running test suites, yet we put `Pre build` task at first, then run the tests, then finally we run the test suite which requires `Pre build` task
* Both `Run ci-asm` and `Run test suite` tasks contains installation commands involving `apt`. So we cannot say that `Pre build` prepares the environment
* `Pre build` is not only preparing the environments, we are building test suites here.

So if we want to do a re-organization here, I would personally merge `Pre build` task with `Run test suite`, or we should at least move all `apt` related commands to `Pre build` tasks.